### PR TITLE
test_: Tests for wakuext messages 1

### PR DESCRIPTION
**Pull request by rlve**
Functional tests for `wakuext` messages methods: https://github.com/status-im/status-go/issues/6084

Important changes:
- [x] `wakuext_chatMessages`
- [x] `wakuext_messageByMessageID`
- [x] `wakuext_allMessagesFromChatWhichMatchTerm`

### CODE REVIEW

File: tests-functional/tests/test_cases.py [hunk 1]
```diff
@@ -380,6 +380,18 @@ def one_to_one_message(self, message_count):
 
         return responses
 
+    def send_multiple_one_to_one_messages(self, message_count=1) -> tuple[list[str], list[dict]]:
```
Comment (fbarbu15) [inline tests-functional/tests/test_cases.py:377]: could we write this function in such a way that it could be also reused inside `one_to_one_message` 
the code is pretty similar and it would be a shame to duplicate it

File: tests-functional/tests/test_wakuext_messages.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+from tests.test_cases import MessengerTestCase
+
+
+@pytest.mark.usefixtures("setup_two_unprivileged_nodes")
+@pytest.mark.rpc
+class TestChatMessages(MessengerTestCase):
+
+    def test_chat_messages(self):
+        sent_texts, _ = self.send_multiple_one_to_one_messages(1)
```
Comment (fbarbu15) [inline tests-functional/tests/test_wakuext_messages.py:11]: this is somehow missleading, name suggests multiple messages while you are sending just one

File: tests-functional/tests/test_wakuext_messages.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        assert actual_text == sent_texts[0]
+
+    def test_chat_messages_with_pagination(self):
+        sent_texts, _ = self.send_multiple_one_to_one_messages(5)
+        sender_chat_id = self.receiver.public_key
+
+        # Page 1
+        chat_messages_res1 = self.sender.wakuext_service.chat_messages(sender_chat_id, cursor="", limit=3)
+
+        cursor1 = chat_messages_res1.get("result", {}).get("cursor", "")
```
Comment (fbarbu15) [inline tests-functional/tests/test_wakuext_messages.py:30]: should we have an asertion that cursor1 in not empty?

File: tests-functional/tests/test_wakuext_messages.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        response = self.sender.wakuext_service.chat_messages(sender_chat_id)
+
+        self.sender.verify_json_schema(response, method="wakuext_chatMessages")
+
+        messages = response.get("result", {}).get("messages", [])
+        assert len(messages) == 1
+        actual_text = messages[0].get("text", "")
+        assert actual_text == sent_texts[0]
+
+    def test_chat_messages_with_pagination(self):
```
Comment (fbarbu15) [inline tests-functional/tests/test_wakuext_messages.py:23]: is there a max page size where we could test boundaries?

**Review by fbarbu15 — COMMENTED** [review]
thanks, left few questions/comments

File: tests-functional/tests/test_cases.py (hunk 1 above)
Comment (rlve) [reply tests-functional/tests/test_cases.py:377]: done in https://github.com/status-im/status-go/pull/6356/commits/5006a236f2e0eaccf29ef603a6ce303d704d2e43

File: tests-functional/tests/test_wakuext_messages.py (hunk 2 above)
Comment (rlve) [reply tests-functional/tests/test_wakuext_messages.py:11]: I don't agree, the name of the function relates to its capability to send multiple `one_to_one` messages, which is exactly what it can do. 

In this case I'm using it to send only one message, as I don't need more.

File: tests-functional/tests/test_wakuext_messages.py (hunk 3 above)
Comment (rlve) [reply tests-functional/tests/test_wakuext_messages.py:30]: It's tested implicitly when it's used in line https://github.com/status-im/status-go/pull/6356/files#diff-e4a860b3db9ee93326b83b4cb1f008b7acf553f59a9bd4dc5469575a920587feR38.

But sure, I can add an assertion.

File: tests-functional/tests/test_wakuext_messages.py (hunk 4 above)
Comment (rlve) [reply tests-functional/tests/test_wakuext_messages.py:23]: It seems that there is no explicit limit, at least I can't find it in the code.

File: tests-functional/tests/test_wakuext_messages.py (hunk 2 above)
Comment (fbarbu15) [reply tests-functional/tests/test_wakuext_messages.py:11]: ok, makes sense

File: tests-functional/tests/test_wakuext_messages.py (hunk 3 above)
Comment (fbarbu15) [reply tests-functional/tests/test_wakuext_messages.py:30]: oh right, then no need

File: tests-functional/tests/test_wakuext_messages.py (hunk 4 above)
Comment (fbarbu15) [reply tests-functional/tests/test_wakuext_messages.py:23]: I think the pagination is on the waku side, something like 100 should be max IIRC

Comment (fbarbu15) [reply tests-functional/tests/test_wakuext_messages.py:23]: But I guess we don't necessarily need such edge cases at this point

**Review by fbarbu15 — APPROVED** [review]

File: tests-functional/tests/test_wakuext_messages.py (hunk 4 above)
Comment (rlve) [reply tests-functional/tests/test_wakuext_messages.py:23]: I played around with it and it was possible to retrieve even 1500 messages.

**Review by antdanchenko — APPROVED** [review]
